lib/plugin/instance.rb

```diff
@@ -160,11 +160,11 @@ def replace_flags(settings: ::FlagSettings.new, score_type_names: [])
     end
   end
 
-  def whitelist_staff_user_custom_field(field)
+  def allow_staff_user_custom_field(field)
     DiscoursePluginRegistry.register_staff_user_custom_field(field, self)
   end
 
-  def whitelist_public_user_custom_field(field)
+  def allow_public_user_custom_field(field)
```

Comment:
This is a plugin API, I don't think we should rename it without a deprecation notice. Can we keep the old name working (with deprecation warning)? It can just call the new function.